dask/_task_spec.py

```diff
@@ -790,3 +789,87 @@ def execute_graph(
                 del cache[dep]
 
     return cache
+
+
+def fuse_linear_task_spec(dsk, keys):
+    """
+    keys are the keys from the graph that are requested by a computation. We
+    can't fuse those together.
+    """
+    from dask.core import reverse_dict
+    from dask.optimization import default_fused_keys_renamer
+
+    keys = set(keys)
+    dependencies = DependenciesMapping(dsk)
+    dependents = reverse_dict(dependencies)
+
+    seen = set()
+    result = {}
+
+    def _check_data_node(tsk):
```

Comment:
@fjetter and I chatted about this and we don't see any benefit of not fusing those tasks. I'd still prefer to do this as a follow up because they will break some tests which makes review harder than needed